pageserver/src/walingest.rs

```diff
@@ -403,14 +404,26 @@ impl<'a> WalIngest<'a> {
             && decoded.xl_rmid == pg_constants::RM_XLOG_ID
             && (decoded.xl_info == pg_constants::XLOG_FPI
                 || decoded.xl_info == pg_constants::XLOG_FPI_FOR_HINT)
-        // compression of WAL is not yet supported: fall back to storing the original WAL record
-            && !postgres_ffi::bkpimage_is_compressed(blk.bimg_info, self.timeline.pg_version)?
+        // only lz4 compression of WAL is now supported, for other cmpression algorithms fall back to storing the original WAL record
+            && (!postgres_ffi::bkpimage_is_compressed(blk.bimg_info, self.timeline.pg_version)? ||
```

Comment:
Postgres15 supports zstd compression since v15 (at least, it was mentioned about basebackup, but I'm not sure about WAL).

Reply:
 Yes, PG15 supports lz4 and zest wal compression